sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/interface.scala

```diff
@@ -61,36 +61,60 @@ trait MetadataMapSupport {
     jsonToString(toJsonLinkedHashMap)
   }
 
+  private def jsonToStringReformat(
+      key: String, jValue: JValue, map: mutable.LinkedHashMap[String, String]): Unit = {
+    val reformattedValue: String = if (key == "Statistics") {
+      jValue match {
+        case JObject(fields) =>
+          fields.flatMap {
+            case ("size_in_bytes", JDecimal(bytes)) =>
+              Some(s"$bytes bytes")
+            case ("num_rows", JDecimal(rows)) =>
+              Some(s"$rows rows")
+            case _ => None
+          }.mkString(", ")
+        case _ => jValue.values.toString
+      }
+    } else if (key == "Created Time" || key == "Last Access") {
+      jValue match {
+        case JLong(value) => new Date(value).toString
+        case _ => jValue.values.toString
+      }
+    } else {
+      jValue.values.toString
+    }
+    map.put(key, reformattedValue)
+  }
+
   protected def jsonToString(
       jsonMap: mutable.LinkedHashMap[String, JValue]): mutable.LinkedHashMap[String, String] = {
     val map = new mutable.LinkedHashMap[String, String]()
-    val timestampKeys = Set("Created Time", "Last Access")
+    val reformatKeys = Set("Statistics", "Created Time", "Last Access")
     jsonMap.foreach { case (key, jValue) =>
-      val stringValue = jValue match {
-        case JString(value) => value
-        case JArray(values) =>
-          values.map(_.values)
-            .map {
-              case str: String => quoteIdentifier(str)
-              case other => other.toString
+      if (reformatKeys.contains(key)) {
+        jsonToStringReformat(key, jValue, map)
```

Comment:
how about make `jsonToStringReformat` return an optional key/value, so that we don't need to keep track of `reformatKeys` here.